src/Swarm/TUI/Controller.hs

```diff
@@ -808,27 +816,30 @@ adjustTPS (+/-) = uiState . lgTicksPerSecond %~ (+/- 1)
 ------------------------------------------------------------
 
 -- | Handle user input events in the robot panel.
-handleRobotPanelEvent :: AppState -> BrickEvent Name AppEvent -> EventM Name (Next AppState)
-handleRobotPanelEvent s (Key V.KEnter) =
-  maybe (continueWithoutRedraw s) (descriptionModal s) (focusedEntity s)
-handleRobotPanelEvent s (CharKey 'm') =
-  maybe (continueWithoutRedraw s) (makeEntity s) (focusedEntity s)
-handleRobotPanelEvent s (CharKey '0') = do
-  continue $ s & (uiState . uiShowZero %~ not) . (uiState . uiInventoryShouldUpdate .~ True)
-handleRobotPanelEvent s (VtyEvent ev) = do
-  let mList = s ^? uiState . uiInventory . _Just . _2
+handleRobotPanelEvent :: BrickEvent Name AppEvent -> EventM Name AppState ()
+handleRobotPanelEvent (Key V.KEnter) = do
+  focused <- gets focusedEntity
+  maybe continueWithoutRedraw descriptionModal focused
+handleRobotPanelEvent (CharKey 'm') = do
+  focused <- gets focusedEntity
+  maybe continueWithoutRedraw makeEntity focused
+handleRobotPanelEvent (CharKey '0') = do
+  uiState . uiInventoryShouldUpdate .= True
+  uiState . uiShowZero %= not
+handleRobotPanelEvent (VtyEvent ev) = do
+  mList <- preuse $ uiState . uiInventory . _Just . _2
   case mList of
-    Nothing -> continueWithoutRedraw s
+    Nothing -> continueWithoutRedraw
     Just l -> do
-      l' <- handleListEventWithSeparators ev (is _Separator) l
-      let s' = s & uiState . uiInventory . _Just . _2 .~ l'
-      continue s'
-handleRobotPanelEvent s _ = continueWithoutRedraw s
+      l' <- nestEventM' l (handleListEventWithSeparators ev (is _Separator))
+      uiState . uiInventory . _Just . _2 .= l'
```

Comment:
Could this whole thing somehow be done more simply with `zoom (uiState . uiInventory . _Just . _2)`?

Reply:
I think not, we want to continue without redrawing the screen in the other case.

If it was just `return ()` then `zoom` would work.